upgrade/upgrade-scripts/upgrade

```diff
@@ -361,6 +361,41 @@ function rollback() {
 }
 
 function finalize() {
+	source_upgrade_properties
+
+	[[ -n "$UPGRADE_TYPE" ]] ||
+		die "variable UPGRADE_TYPE is not set; is upgrade in progress?"
+	[[ -n "$UPGRADE_BASE_VERSION" ]] ||
+		die "variable UPGRADE_BASE_VERSION is not set"
+
+	case "$UPGRADE_TYPE" in
+	DEFERRED | FULL | ROLLBACK) ;;
+	*)
+		die "finalize is not supported for upgrade type: '$UPGRADE_TYPE'"
+		;;
+	esac
+
+	remove_upgrade_properties
+
+	#
+	# We only want to remove this hotfix file after a successful
+	# upgrade attempt; at which point the old hotfix file will no
+	# longer be representative of the new rootfs contents.
+	#
+	# If a rollback occured, then we do not want to remove this
+	# file, since the currently running rootfs will be the "old"
+	# version, and still should match the hotfix file. We detect if
+	# a rollback occurred by the presence of this field (which would
+	# be set in the upgrade properites file).
+	#
+	if [[ -z "$ROLLBACK_BASE_VERSION" ]]; then
```

Comment:
I agree with this location for removing the hotfix file. It is critical to remove it similar to removing the upgrade.properties file once we are beyond the point of no return as far as upgrade is concerned. So lets keep it here in finalize().